doc/function_types.md

```diff
@@ -841,6 +841,47 @@ int main()
 
 </details>
 
+
+## Multi-objective functions
+
+A multi-objective function is a set of functions or objectives that are
+simultaneously optimized to reach an optimum.
```

Comment:
Could we also add something to this quick description like this?

```
A multi-objective optimizer does not return just one set of coordinates at the
minimum of all objective functions, but instead finds a *front* or *frontier* of
possible coordinates that are Pareto-optimal (that is, no individual objective
function's value can be reduced without increasing at least one other
objective function).
```

I don't know if that's the best description, feel free to change it. :+1:
```

Reply:
Sure, on it!